File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/firebase/firestore/transactions/user/account/FirestoreAccount.java
package com.TritiumGaming.phasmophobiaevidencepicker.firebase.firestore.transactions.user.account;

import androidx.annotation.NonNull;

import com.TritiumGaming.phasmophobiaevidencepicker.firebase.firestore.transactions.user.FirestoreUser;
import com.TritiumGaming.phasmophobiaevidencepicker.firebase.firestore.transactions.user.account.transactions.FirestoreTransactionHistory;
import com.google.firebase.firestore.CollectionReference;

public class FirestoreAccount {

    public final static String COLLECTION_ACCOUNT = "Account";

    public static void init()
            throws Exception {
        FirestoreAccountCredit.init();
        FirestoreTransactionHistory.init();
    }

    @NonNull
    public static CollectionReference getAccountCollection()
            throws Exception {
        return FirestoreUser.getUserDocument().collection(COLLECTION_ACCOUNT);
    }

}

File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/firebase/firestore/transactions/user/account/FirestoreAccount.kt
package com.TritiumGaming.phasmophobiaevidencepicker.firebase.firestore.transactions.user.account

import com.TritiumGaming.phasmophobiaevidencepicker.firebase.firestore.transactions.user.FirestoreUser.Companion.userDocument
import com.TritiumGaming.phasmophobiaevidencepicker.firebase.firestore.transactions.user.account.transactions.FirestoreTransactionHistory
import com.TritiumGaming.phasmophobiaevidencepicker.firebase.firestore.transactions.user.account.properties.FirestoreAccountCredit
import com.google.firebase.firestore.CollectionReference

class FirestoreAccount {
    companion object {
        private const val COLLECTION_ACCOUNT: String = "Account"

        @Throws(Exception::class)
        fun init() {
            FirestoreAccountCredit.init()
            FirestoreTransactionHistory.init()
        }

        val accountCollection: CollectionReference
            @Throws(Exception::class)
            get() = userDocument.collection(COLLECTION_ACCOUNT)
    }
}
